src/contact/parallel/bvh_contact_manager.cc

```diff
@@ -0,0 +1,279 @@
+      vis_file << "Contact entity visualization" << std::endl;
+
+      // file format ASCII | BINARY
+      vis_file << "ASCII" << std::endl;
+
+      // dataset structure STRUCTURED_POINTS | STRUCTURED_GRID | UNSTRUCTURED_GRID | POLYDATA | RECTILINEAR_GRID | FIELD
+      vis_file << "DATASET UNSTRUCTURED_GRID" << std::endl;
+
+      std::vector< std::vector<int> > vtk_triangles(faces.size());
+
+      vis_file << "POINTS " << 3*faces.size() << " float" << std::endl;
+
+      for (int i=0 ; i<faces.size() ; i++) {
+        ContactEntity const & face = faces[i];
+        vis_file << face.coord_1_x_ << " " << face.coord_1_y_ << " " << face.coord_1_z_ << std::endl;
+        vis_file << face.coord_2_x_ << " " << face.coord_2_y_ << " " << face.coord_2_z_ << std::endl;
+        vis_file << face.coord_3_x_ << " " << face.coord_3_y_ << " " << face.coord_3_z_ << std::endl;
+        vtk_triangles[i].push_back(3*i);
+        vtk_triangles[i].push_back(3*i + 1);
+        vtk_triangles[i].push_back(3*i + 2);
+      }
+
+      vis_file << "CELLS " << vtk_triangles.size() << " " << 4*vtk_triangles.size() << std::endl;
+      for(const auto & vtk_triangle : vtk_triangles) {
+        vis_file << "3 " << vtk_triangle[0] << " " << vtk_triangle[1] << " " << vtk_triangle[2] << std::endl;
+      }
+
+      vis_file << "CELL_TYPES " << faces.size() << std::endl;
+      for(unsigned int i=0 ; i<vtk_triangles.size() ; ++i) {
+        // cell type 6 is VTK_TRIANGLE
+        vis_file << 6 << std::endl;
+      }
+
+      vis_file.close();
+    }
+
+    void
+    WriteContactEntitiesToVTKFile(const std::vector<ContactEntity> &faces,
+                                  const std::vector<ContactEntity> &nodes,
+                                  const std::string &prefix,
+                                  int step) {
+
+      std::stringstream file_name_ss;
+      file_name_ss << prefix;
+      file_name_ss << std::setfill( '0' ) << std::setw( 5 ) << step;
+      file_name_ss << ".vtk";
+      std::ofstream vis_file;
+      vis_file.open(file_name_ss.str().c_str());
+
+      // file version and identifier
+      vis_file << "# vtk DataFile Version 3.0" << std::endl;
+
+      // header
+      vis_file << "Contact entity visualization" << std::endl;
+
+      // file format ASCII | BINARY
+      vis_file << "ASCII" << std::endl;
+
+      // dataset structure STRUCTURED_POINTS | STRUCTURED_GRID | UNSTRUCTURED_GRID | POLYDATA | RECTILINEAR_GRID | FIELD
+      vis_file << "DATASET UNSTRUCTURED_GRID" << std::endl;
+
+      std::vector<int> vtk_vertices(nodes.size());
+      std::vector< std::vector<int> > vtk_triangles(faces.size());
+
+      vis_file << "POINTS " << nodes.size() + 3*faces.size() << " float" << std::endl;
+
+      for (int i=0 ; i<nodes.size() ; i++) {
+        vis_file << nodes[i].coord_1_x_ << " " << nodes[i].coord_1_y_ << " " << nodes[i].coord_1_z_ << std::endl;
+        vtk_vertices[i] = i;
+      }
+      int offset = static_cast<int>(vtk_vertices.size());
```

Comment:
For large models you could consider using the base64 format encoding provided by the VTK library itself.